internal/cmd/gocdk/internal/docker/docker.go

```diff
@@ -0,0 +1,228 @@
+// API directly in the future.
+package docker
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"os/exec"
+	"strings"
+	"time"
+
+	"golang.org/x/xerrors"
+)
+
+// Client is a client for a Docker daemon. The zero value is a Docker client
+// that uses the current process's environment. Methods on Client are safe to
+// call concurrently from multiple goroutines.
+type Client struct {
+	env []string
+}
+
+// New creates a new client with the given environment variables.
+// Calling docker.New(nil) is the same as new(docker.Client) and uses the
+// current process's environment.
+func New(env []string) *Client {
+	return &Client{env: env}
+}
+
+// Image stores metadata about a Docker image.
+type Image struct {
+	// ID is the randomly generated ID of the image.
+	// See https://windsock.io/explaining-docker-image-ids/
+	ID string
+	// Repository is the name component of a Docker image reference.
+	// It may be empty if the image has no tags.
+	// See https://godoc.org/github.com/docker/distribution/reference
+	Repository string
+	// Tag is the tag component of a Docker image reference.
+	// It will be empty if the image has no tags.
+	// See https://godoc.org/github.com/docker/distribution/reference
+	Tag string
+	// Digest is the content-based hash of the image.
+	// It may be empty.
+	Digest string
+	// CreatedAt is the time the image was built.
+	CreatedAt time.Time
+}
+
+// ListImages lists images the Docker daemon has stored locally that match filterRef.
+func (c *Client) ListImages(ctx context.Context, filterRef string) ([]*Image, error) {
+	args := []string{"images", "--digests", "--format={{json .}}"}
+	if filterRef != "" {
+		args = append(args, filterRef)
+	}
+	cmd := exec.CommandContext(ctx, "docker", args...)
+	cmd.Env = c.env
+	out, err := run(cmd)
+	if err != nil {
+		return nil, xerrors.Errorf("list docker images: %w", err)
+	}
+	lf := []byte("\n")
+	out = bytes.TrimSuffix(out, lf)
+	var images []*Image
+	for _, line := range bytes.Split(out, lf) {
+		var img struct {
+			ID         string
+			Repository string
+			Tag        string
+			Digest     string
+			CreatedAt  string
+		}
+		if err := json.Unmarshal(line, &img); err != nil {
+			return nil, xerrors.Errorf("list docker images: parse output: %w", err)
+		}
+		createdAt, _ := time.Parse("2006-01-02 15:04:05 -0700 MST", img.CreatedAt)
+		images = append(images, &Image{
+			ID:         cleanNone(img.ID),
+			Repository: cleanNone(img.Repository),
+			Tag:        cleanNone(img.Tag),
+			Digest:     cleanNone(img.Digest),
+			CreatedAt:  createdAt,
+		})
+	}
+	return images, nil
+}
+
+func cleanNone(s string) string {
+	if s == "<none>" {
+		return ""
+	}
+	return s
+}
+
+// RunOptions holds optional parameters for starting a Docker container.
+type RunOptions struct {
+	Args         []string
+	Env          []string
+	RemoveOnExit bool
+	Publish      []string
+}
+
+// Start starts a Docker container. It does not wait for the container to stop.
```

Comment:
As a reader of this documentation comment, I am very confused about the `It does not wait for the container to stop.` Should I be worried about it waiting for something to stop?/I don't understand why it's telling me this.

Reply:
Removed the sentence.